__tests__/app.test.js

```diff
@@ -0,0 +1,38 @@
+const app = require('../app')
+const db = require('../db/connection')
+const seed = require('../db/seeds/seed')
+const data = require('../db/data/test-data/index')
+const request = require('supertest')
+
+
+afterAll(() => {
+    db.end()
+   })
+
+beforeEach(() => {
+return seed(data)
+})
+
+
+describe("GET /api/topics",()=>{
+
+    test("200: /api/topics should return all topics available",()=>{
+
+        return request(app).get('/api/topics').expect(200)
+        .then(({body})=>{
+            body.allTopics.forEach((topic)=>{
+                expect(topic).toHaveProperty("slug", expect.any(String));
+                expect(topic).toHaveProperty("description", expect.any(String));
+
+            })
+        })
+    })
```

Comment:
Hi Gary - we need to be very careful when writing expect statements within a forEach. If we get back an empty array from this request this forEach would never run and so the expect statements would never run, giving us an auto pass. We need an expect statement before the iteration to guard against this, how could we write this?